### PR TITLE
src/test: fixups for kernels without ACL and net-dummy

### PR DESCRIPTION
**Pull request by MaxKellermann**
Skip tests that require ACL or net-dummy support on kernels that have these features disabled. This avoids unnecessary unit test failures on such kernels.

### CODE REVIEW

**Comment by MaxKellermann** [comment]
Check "ci/semaphoreci/pr: Debian autopkgtest (LXC)" failure is unrelated to this PR.

**Comment by MaxKellermann** [comment]
Same for the two "jammy" tests:
```
110s Cloning into 'systemd'...
111s fatal: unable to access 'https://salsa.debian.org/systemd-team/systemd.git/': The requested URL returned error: 503
126s Cloning into 'systemd'...
126s fatal: unable to access 'https://salsa.debian.org/systemd-team/systemd.git/': The requested URL returned error: 503
```

File: src/test/test-chown-rec.c [hunk 1]
```diff
@@ -95,7 +96,11 @@ TEST(chown_recursive) {
 
         /* We now apply an xattr to the dir, and check it again */
         p = strjoina(t, "/dir");
-        assert_se(setxattr(p, "system.posix_acl_access", acl, sizeof(acl), 0) >= 0);
+        r = setxattr(p, "system.posix_acl_access", acl, sizeof(acl), 0);
```
Comment (bluca) [inline src/test/test-chown-rec.c]: `r = RET_NERRNO(setxattr(`

File: src/test/test-chown-rec.c [hunk 2]
```diff
@@ -95,7 +96,11 @@ TEST(chown_recursive) {
...
+        if (r < 0 && ERRNO_IS_NOT_SUPPORTED(errno))
```
Comment (bluca) [inline src/test/test-chown-rec.c]: `if (ERRNO_IS_NEG_NOT_SUPPORTED(r))`

File: src/test/test-execute.c [hunk 3]
```diff
@@ -1074,6 +1075,11 @@ static void test_exec_ambientcapabilities(Manager *m) {
 static void test_exec_privatenetwork(Manager *m) {
         int r;
 
+        if (!have_net_dummy) {
+                log_notice("Skipping %s, dummy network interface not available", __func__);
```
Comment (bluca) [inline src/test/test-execute.c]: `return (void) log_notice(`

**Review by bluca — APPROVED** [review]

File: src/test/test-execute.c (hunk 3 above)
Comment (MaxKellermann) [reply src/test/test-execute.c]: Changed - I saw both code styles existed, but in this very function, it was not used that way.

**Comment by MaxKellermann** [comment]
Thanks @bluca, I force-pushed with your suggested changes.

File: src/test/test-chown-rec.c [hunk 4]
```diff
@@ -95,7 +96,11 @@ TEST(chown_recursive) {
 
         /* We now apply an xattr to the dir, and check it again */
         p = strjoina(t, "/dir");
-        assert_se(setxattr(p, "system.posix_acl_access", acl, sizeof(acl), 0) >= 0);
+        r = RET_NERRNO(setxattr(p, "system.posix_acl_access", acl, sizeof(acl), 0));
+        if (ERRNO_IS_NEG_NOT_SUPPORTED(r))
+                return (void) log_tests_skipped_errno(errno, "no acl supported on /tmp");
```
Comment (bluca) [inline src/test/test-chown-rec.c]: `log_tests_skipped_errno(r`

Comment (MaxKellermann) [reply src/test/test-chown-rec.c]: D'oh!

**Pull request merged**